.github/CONTRIBUTING.md

```diff
@@ -50,192 +48,260 @@ npm run bootstrap
+1. In a new terminal, run `npm test` to start Jest. When Jest runs the test pages, it will take screenshots of new tests and save it under [`/__tests__/__image_snapshots__/html`](https://github.com/microsoft/BotFramework-WebChat/tree/main/__tests__/__image_snapshots__/html)
 
-There are 2 ways to run the test suite, either one-off or continuously:
+### Troubleshooting test suites
 
-> To run Web Driver test, locally, download [ChromeDriver](https://sites.google.com/a/chromium.org/chromedriver/downloads) and extract to project root. Then set environment variable `WEBCHAT_TEST_ENV=chrome-local` before running Jest.
+We run test suites on every PR and require 100% pass rate. If test suites do not complete successfully, the cause may be:
 
--  For one-off testing, `node_modules/.bin/jest`
--  For continuous testing, `npm test`
-   -  To speed up continuous testing, you can install [watchman](https://facebook.github.io/watchman/)
+- New changes are causing failure(s) in existing tests
+- Intermittent services issues
+- Test reliability issue, please see [#2938](https://github.com/microsoft/BotFramework-WebChat/issues/2938)
+- Polluted development environment, for example:
+  - Outdated `node_modules` content
+  - Outdated Node.js or NPM
 
-If your development box has less than 4 cores, you will need to reduce the number of simultaneous test agents:
+When the test suites fail:
 
--  For one-off testing: run `node_modules/.bin/jest --maxWorkers 1`
--  For continuous testing: run `npm test -- --maxWorkers 1`
+- Identify whether the tests fail WITHOUT any code changes
+  - Make a fresh clone of BotFramework-WebChat and run the test suites without any changes. If the fresh clone tests pass, this means the latest code changes are causing failures.
+- Identify whether the failure is intermittent
+  - Run the (failing) test suites again, potentially using Jest filters (<kbd>F</kbd> for failed tests)
 
-Our CI pipeline run tests with 4 agents simultaneously. If new tests are added, please make sure they can run simultaneously.
+If existing test suites fail without any code changes, please determine the following:
 
-### Troubleshooting the test suite
+- Test suites always fail, even after repeated runs
+  1.  The service may have been updated, causing the test suite failures. Please [file an issue](https://github.com/microsoft/BotFramework-WebChat/issues/new/choose) on the Web Chat repository.
+- Test suites fail intermittently
+  1.  Service reliability problems (e.g. for DirectLine, or Mockbot) may be the cause. Please [file an issue](https://github.com/microsoft/BotFramework-WebChat/issues/new/choose) on the Web Chat repository.
+  1.  Test reliability problems may be the cause. If so, please comment on [#2938](https://github.com/microsoft/BotFramework-WebChat/issues/2938) and include:
+  - Failing test names/files
+  - Failing screencaps, if available. These images can be retrieved from `__diff_output__`
+  - Error messages
```

Comment:
I think there's a spacing issue here - bullets are in the same level as the parent ordered list. Maybe one more space per line is needed?

Reply:
![image](https://user-images.githubusercontent.com/14900841/118196964-ce5bf780-b402-11eb-9b77-f4228cf0916b.png)

Reply:
Fixed.